code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/builders/AepPushNotificationBuilder.kt

```diff
@@ -0,0 +1,974 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.services.ui.notification.builders
+
+import android.app.Activity
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.BroadcastReceiver
+import android.content.ContentResolver
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.graphics.Bitmap
+import android.graphics.Color
+import android.media.RingtoneManager
+import android.net.Uri
+import android.os.Build
+import android.view.View
+import android.widget.RemoteViews
+import androidx.annotation.RequiresApi
+import androidx.core.app.NotificationCompat
+import com.adobe.marketing.mobile.MobileCore
+import com.adobe.marketing.mobile.core.R
+import com.adobe.marketing.mobile.services.Log
+import com.adobe.marketing.mobile.services.ServiceProvider
+import com.adobe.marketing.mobile.services.ui.notification.NotificationConstructionFailedException
+import com.adobe.marketing.mobile.services.ui.notification.PushTemplateConstants
+import com.adobe.marketing.mobile.services.ui.notification.PushTemplateImageUtil
+import com.adobe.marketing.mobile.services.ui.notification.models.AepPushTemplate
+import com.adobe.marketing.mobile.services.ui.notification.models.ManualCarouselPushTemplate
+import com.adobe.marketing.mobile.util.UrlUtils
+import java.util.Random
+
+internal object AepPushNotificationBuilder {
```

Comment:
We can do this as a followup task but adding a comment here so that we keep track of it. 
The utilities provided by this builder _assumes_ the id's for various common elements (R.id.basic_small_layout , R.id.notification_title, R.id.notification_body_expanded) are the same across templates. We will need to figure out a way to enforce this somehow either programmatically, structurally in the layout  or via documentation.
Can you add a TODO and a note here for this please.